### PR TITLE
Fixed OSX/Darwin link status property and parsing

### PR DESCRIPTION
**Pull request by pmvenegas**
This will break code that relies on NetworkAdapter#status (incorrectly)
referring to flags instead of link status.

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 $:.unshift File.expand_path('../lib', File.dirname(__FILE__))
 require 'ifconfig'
-require 'test/unit'
+require 'minitest/autorun'
 require 'pp'
+
+Test = Minitest
```
Comment (asomers) [inline test/test_helper.rb:6]: What's the purpose of the changes to test/test_helper.rb?  They seem unrelated to the rest of the pull request.

Comment (pmvenegas) [inline test/test_helper.rb:6]: Sorry, this was to run unit tests on my local machine. Feel free to drop changes to test_helper.rb.

**Comment by asomers** [comment]
The unit tests don't pass for me when I check out your branch.  I get this error on ruby 1.9.3.  On ruby 2.1.6 the error is formatted differently, but it's still the same assertion.

$ rake test
Loaded suite /usr/local/lib/ruby/gems/1.9/gems/rake-10.0.4/lib/rake/rake_test_loader
Started
....F..................................
Finished in 0.279701 seconds.

  1) Failure:
test_link_status(TC_DarwinTest) [/usr/home/alans/src/ruby-ifconfig/test/unit/tc_darwin.rb:43]:
Failed to parse link status

39 tests, 53 assertions, 1 failures, 0 errors, 0 skips

Test run options: --seed 15371
rake aborted!
Command failed with status (1): [ruby -I"lib" -I"/usr/local/lib/ruby/gems/1.9/gems/rake-10.0.4/lib" "/usr/local/lib/ruby/gems/1.9/gems/rake-10.0.4/lib/rake/rake_test_loader.rb" "test/unit/tc_darwin.rb" "test/unit/tc_dragonflybsd.rb" "test/unit/tc_freebsd.rb" "test/unit/tc_netbsd.rb" "test/unit/tc_openbsd.rb" "test/unit/tc_osx.rb" ]

Tasks: TOP => test => test:bsd
(See full trace by running task with --trace)

**Comment by asomers** [comment]
Closing due to lack of feedback from the submitter.
